docs/README.md

```diff
@@ -1,9 +1,131 @@
+
+
+[MS-RDPBCGR]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/5073f4ed-1e93-45e1-b039-6e30c385867c
+[connection]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/023f1e69-cfe8-4ee6-9ee0-7e759fb4e4ee
+
+### High Level Architecture
+
+PyRDP is split into several modules. This section attempts to give a
+high level view of the purpose of each module and how they interact
+with one another.
+
+The **core** module contains essential code that is used across the
+entire project. These are stand-alone primitives and helper functions
+that are highly reusable. Examples include:
+
+- BER Encoding
+- Configuration File Management
+- Stream Parsing and Packing Helpers
+- Twisted Primitives
+- Base Types, Abstract Classes and Interfaces
+
+
+The **enum** module exposes constants that are proper to the RDP
+protocol and not internal to a specific module. These include:
+
+- Keyboard scancodes
+- RDP message types
+- Bitfield flags specific to messages and protocol data units (PDUs)
+
+
+The **layer** module contains the core of the PyRDP networking
+stack. A [separate document][layers] explains the data flow through
+layers and is essential to understanding how the data flows between
+the client connection and the server connection. Each protocol is
+separated into its own layer class. The layer mechanism is also used
+to abstract recording to disk.
+
+
+The **logging** module offers support for uniform logging throughout
+the PyRDP project. It provides custom formatters and primitives for
+filtering log output.
+
+
+The **mcs** module provides abstractions for RDP data channels.
+
+
+The **mitm** module is the main component responsible for connection
+interception and tampering. It contains interception facilities for
+each layer of the protocol and hooks to inspect and modify data.  It
+also contains the application logic for the MITM server and
+establishing connectivity with the target server.
+
+
+The **pdu** module centralizes classes to represent all of the
+supported RDP message types. These PDUs can be read and written from a
+connection and are the main view into the RDP data stream. Example PDUs
+include:
+
+- DemandActive
+- ConfirmActive
+- ClientInfo
+
+
+The **parser** module is the heart of the RDP implementation. This is
+where the wire protocol is converted from raw bytes into PDUs and from
+PDUs back into raw bytes. Parsers are split according to the protocol
+layer they operate on and follow the same logical structure as the RDP
+specification. Here are some example parsers:
+
+- SlowPathParser
+- FastPathParser
+- BitmapParser
+
+
+The **player** module contains code and logic for the Player component
+to render an actual RDP session into various forms. The most common
+form is to render to the player window, but other types of rendering
+are possible (text-only, for example) and can be implemented here.
+
+
+The **recording** module offers capabilities to export an RDP session
```

Comment:
Not just disk, over the network as well

Reply:
The "sending over the network" part is not happening in the recording module. The recorder must be given a PDU and it passes it into its player layers blindly without caring where the data ends up, which may or may not include network layers, but is currently exclusively a FileLayer in the codebase (AFAICT.)

In fact, while I was reading the recorder code to reply to your comment, I realized that the whole recorder could have just been be an extra layer that composes/subclasses the PlayerLayer. It's too late to refactor it and not really worth it, though. The actual recording is done by the `FileLayer`. I still think that for documentation clarity, the main use case of the module should be documented. If you don't agree, feel free to provide the description you would like to see for the recording module, and I'll update the text to include it.